src/box/decimal.h

```diff
@@ -515,19 +515,22 @@ box_decimal_mp_decode(box_decimal_t *dec, const char **data);
  * Decode a decimal from msgpack @a data without the msgpack
  * extension header.
  *
- *  | box_decimal_mp_decode_data() must be called for this position
- *  |                                        |
- *  |                                        v
- *  | <msgpack type> <size> <extension type> <data>
- *  | ^
- *  | |
- *  | box_decimal_mp_decode() must be called for this position
+ * \code
+ *  box_decimal_mp_decode_data() must be called for this position
+ *                                         |
+ *                                         v
+ *  <msgpack type> <size> <extension type> <data>
+ *  ^
+ *  |
+ *  box_decimal_mp_decode() must be called for this position
+ * \endcode
  *
  * This function is suitable to finish decoding after calling
  * mp_decode_extl() (from the msgpuck library).
  *
  * @param dec where to store the decoded decimal
  * @param data pointer to a buffer with the msgpack data
+ * @param size size of a buffer a buffer with the msgpack data
```

Comment:
The important detail here is that a caller should pass an exact size of the decimal data, not just size of the buffer. It is the size that `mp_decode_extl()` returns.

I would add this information into the description, because otherwise it is not clear how to call the function. For example:

```c
 * @param size size of the decimal data in the buffer; this value
 *             is returned by mp_decode_extl()
```